src/drivers/bmp280/bmp280_spi.cpp

```diff
@@ -87,10 +87,10 @@ bmp280::IBMP280 *bmp280_spi_interface(uint8_t busnum, uint8_t device, bool exter
 	return new BMP280_SPI(busnum, device, external);
 }
 
-BMP280_SPI::BMP280_SPI(uint8_t bus, uint32_t device, bool external) :
+BMP280_SPI::BMP280_SPI(uint8_t bus, uint32_t device, bool _is_external) :
 	SPI("BMP280_SPI", nullptr, bus, device, SPIDEV_MODE3, 10 * 1000 * 1000)
 {
-	_external = external;
+	_external = _is_external;
```

Comment:
Thanks! Could you please change the variable name to is_external? Variables with underscore should be only class members.

Reply:
Got it, sure... give me a sec... No, it could not be possible due to compiler attempting to reference that name to method *is_external*, but naming as *is_external_device* is ok.

```
[1/5] Building CXX object src/drivers/bmp280/CMakeFiles/drivers__bmp280.dir/bmp280_spi.cpp.obj
FAILED: /usr/bin/arm-none-eabi-g++   -DBUILD_URI=localhost -DCONFIG_ARCH_BOARD_PX4FMU_V2 -DMODULE_NAME=\"bmp280\" -DPX4_MAIN=bmp280_app_main -D__DF_NUTTX -D__PX4_NUTTX -D__STDC_FORMAT_MACROS -I. -Isrc -Isrc/modules -I../../src -I../../src/drivers/boards/px4fmu-v2 -I../../src/include -I../../src/lib -I../../src/lib/DriverFramework/framework/include -I../../src/lib/matrix -I../../src/modules -I../../src/platforms -INuttX/nuttx/arch/arm/src/armv7-m -INuttX/nuttx/arch/arm/src/chip -INuttX/nuttx/arch/arm/src/common -INuttX/nuttx/include -INuttX/nuttx/include/cxx -I../../platforms/nuttx/NuttX/apps/include -Iexternal/Install/include -fno-common -ffunction-sections -fdata-sections -mcpu=cortex-m4 -mthumb -march=armv7e-m -mfpu=fpv4-sp-d16 -mfloat-abi=hard -fno-common -ffunction-sections -fdata-sections -g -fno-exceptions -fno-rtti -std=gnu++11 -fno-threadsafe-statics -DCONFIG_WCHAR_BUILTIN -D__CUSTOM_FILE_IO__ -fcheck-new -Wall -Warray-bounds -Wdisabled-optimization -Werror -Wextra -Wfatal-errors -Wfloat-equal -Wformat-security -Winit-self -Wlogical-op -Wmissing-declarations -Wmissing-field-initializers -Wpointer-arith -Wshadow -Wuninitialized -Wunknown-pragmas -Wunused-variable -Wno-unused-parameter -Wunused-but-set-variable -Wformat=1 -Wdouble-promotion -Wno-missing-field-initializers -Wno-overloaded-virtual -Wreorder -fvisibility=hidden -include visibility.h -fno-strict-aliasing -fomit-frame-pointer -funsafe-math-optimizations -ffunction-sections -fdata-sections -fno-strength-reduce -fno-builtin-printf -Os -DNDEBUG   -fno-strict-aliasing -fomit-frame-pointer -funsafe-math-optimizations -ffunction-sections -fdata-sections -fno-strength-reduce -fno-builtin-printf -Wframe-larger-than=1024 -MMD -MT src/drivers/bmp280/CMakeFiles/drivers__bmp280.dir/bmp280_spi.cpp.obj -MF src/drivers/bmp280/CMakeFiles/drivers__bmp280.dir/bmp280_spi.cpp.obj.d -o src/drivers/bmp280/CMakeFiles/drivers__bmp280.dir/bmp280_spi.cpp.obj -c ../../src/drivers/bmp280/bmp280_spi.cpp
../../src/drivers/bmp280/bmp280_spi.cpp: In constructor 'BMP280_SPI::BMP280_SPI(uint8_t, uint32_t, bool)':
../../src/drivers/bmp280/bmp280_spi.cpp:90:70: error: declaration of 'is_external' shadows a member of 'this' [-Werror=shadow]
 BMP280_SPI::BMP280_SPI(uint8_t bus, uint32_t device, bool is_external) :
                                                                      ^
compilation terminated due to -Wfatal-errors.
cc1plus: all warnings being treated as errors
ninja: build stopped: subcommand failed.
Makefile:153: recipe for target 'px4fmu-v3_default' failed
make: *** [px4fmu-v3_default] Error 1
```